### PR TITLE
Introduce Model in @HensonNavigable annotation

### PR DESCRIPTION
**Pull request by stephanenicolas**
Addresses 
* #67 
* #84

### CODE REVIEW

File: dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java [hunk 1]
```diff
@@ -99,27 +102,7 @@ public void setIsDebugEnabled(boolean isDebugEnabled) {
     this.isDebugEnabled = isDebugEnabled;
   }
 
-  protected Map<TypeElement, InjectionTarget> findAndParseTargets(RoundEnvironment env) {
```
Comment (stephanenicolas) [inline dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java]: Both the InjectExtraProcessor and HensonProcessor were using this code. But indeed InjectExtraProcessor didn't need to parse @HensonNavigable Annotations.

File: dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java [hunk 2]
```diff
@@ -134,19 +117,23 @@ private void parseInjectExtraAnnotatedElements(RoundEnvironment env,
...
       } catch (Exception e) {
         StringWriter stackTrace = new StringWriter();
         e.printStackTrace(new PrintWriter(stackTrace));
 
-        error(element, "Unable to generate extra injector when parsing @Henson.\n\n%s",
+        error(element, "Unable to generate extra injector when parsing @HensonNavigable.\n\n%s",
             stackTrace.toString());
       }
     }
+    for (TypeElement modelTypeElement : modelTypeElements) {
```
Comment (stephanenicolas) [inline dart-common/src/main/java/com/f2prateek/dart/common/AbstractDartProcessor.java]: we remove the model classes from the list of injection targets. We don't want to generate an intent builder for them.

**Comment by dlemures** [comment]
yes you are right

**Comment by dlemures** [comment]
:+1:

**Comment by dlemures** [comment]
:+1:

**Pull request merged**